### ISSUE TITLE
[Android] use additional properties to manage Android version numbers more easily

### ISSUE THREAD

**Issue by wing328** [swagger-api/swagger-codegen#5422]
##### Description

As discussed in https://github.com/swagger-api/swagger-codegen/pull/5250, we want to leverage additional properties to more easily manage Android version numbers in the gradle files generated as part of the Android API client.

We want to replace android version number with {{androidVersionNumber}} tag in the mustache templates so that users can set it directly via `--additional-properties` when generating the API clients.

##### Swagger-codegen version

Latest master

##### Suggest a Fix

If anyone has questions/suggestions or wants to work on this enhancement, please reply to let us know. Thank you.

**Comment by auchri** [comment]
There should be three different variables:
* `{{androidGradleVersion}}` for the version of gradle (used in `classpath 'com.android.tools.build:gradle:2.3.+'`)
* `{{androidSdkVersion}}` for the compile and target sdk varion (used in `compileSdkVersion 25` and `targetSdkVersion 25`)
* `{{androidBuildToolsVersion}}` for the version of the build tools (used in `buildToolsVersion '25.0.2'`)

**Comment by wing328** [comment]
@auchri thanks for the elaboration.

**Comment by cbornet** [comment]
Do we really need to add so much options just to change a version in a gradle file ?

**Comment by wing328** [comment]
@cbornet we can start with just `--additional-properties` to make the versions more easily customizable.

**Comment by YashchenkoN** [comment]
#6160

**Comment by YashchenkoN** [comment]
added #6160 PR

**Comment by wing328** [comment]
Closed via #6160

@auchri please pull the latest master to give it a try.

**Comment by aleksandr1592** [comment]
@wing328 Hello. I'm encountering problem when I want to set to set such properties while building retrofit2 library. Retrofit2 library missing in "android languge" package so I have to build. 
Also at least one more property requried - gradleDistributionUrl or gradleDistributionVersion because gradle android pluging is tightly coupled to gradle version ( see [chart](https://developer.android.com/studio/releases/gradle-plugin.html#updating-gradle))
Do you need request to add such properties to "java language" package ?

**Comment by wing328** [comment]
@aleksandr1592 yes, we would welcome contribution (PR) to add those to the Java (retrofit2) client generator.
